src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -3130,6 +3134,23 @@ internal static bool TryGetStaticEntitySet(this IEdmPathExpression pathExpressio
             return entitySet != null;
         }
 
+        internal static bool HasAny<T>(this IEnumerable<T> enumerable)
+        {
+            IList<T> list = enumerable as IList<T>;
+            if (list != null)
+            {
+                return list.Count() > 0;
+            }
+
+            T[] array = enumerable as T[];
+            if (array != null)
+            {
+                return array.Count() > 0;
+            }
+
+            return enumerable.Any();

```

Comment:
>.Any() [](start = 29, length = 6)

is FirstOrDefault == null better than Any()?

Reply:
Here we can't say the `T` is reference type, or it maybe `Value type`. So, for some scenarios (not a list, not an array), i think it's better to delegate to use "Any()". What do you think? 

---
In reply to: [290009255](https://github.com/OData/odata.net/pull/1473#discussion_r290009255) [](ancestors = 290009255)

Reply:
I'm okay using Any() here.

---
In reply to: [290038532](https://github.com/OData/odata.net/pull/1473#discussion_r290038532) [](ancestors = 290038532,290009255)